core/timer.go

```diff
@@ -0,0 +1,77 @@
+package core
+
+import (
+	"sync"
+	"time"
+
+	"github.com/benbjohnson/clock"
+)
+
+const wakeupWaitTime = 30 * time.Second
+
+// Timer measures active time between start and stop events
+type Timer struct {
+	sync.Mutex
```

Comment:
wozu braucht man den eigentlich? um in mehreren threads die zugriffe im sync zu halten??
macht es sinn die checks und returns vor dem lock zu machen?

Reply:
> wozu braucht man den eigentlich? um in mehreren threads die zugriffe im sync zu halten??
Genau
> macht es sinn die checks und returns vor dem lock zu machen?

Auf keinen Fall, hatte ich geändert. So wie der Code jetzt ist sind alle Zugriffe synchron, könnte also raus, schadet aber auch nichts.